src/calibration.rs

```diff
@@ -0,0 +1,88 @@
+//! Factory-programmed calibration data
+//!
+//! The STM32L0 contains a few read-only registers with factory calibration
+//! data that are written during production.
+
+macro_rules! define_ptr_type {
+    ($name: ident, $ptr: expr) => {
+        impl $name {
+            fn ptr() -> *const Self {
+                $ptr as *const _
+            }
+
+            /// Returns a wrapped reference to the value in flash memory
+            pub fn get() -> &'static Self {
+                unsafe { &*Self::ptr() }
+            }
+        }
+    };
+}
+
+/// Internal voltage reference (V_REFINT)
+///
+/// The internal voltage reference provides a stable (bandgap) voltage output
+/// for the ADC and Comparators. The precise voltage of V_REFINT is
+/// individually measured for each part by ST during production test and stored
+/// in the system memory area. It is accessible in read-only mode.
+#[derive(Debug)]
+#[repr(C)]
+pub struct VrefintCal(u16);
+define_ptr_type!(VrefintCal, 0x1FF8_0078);
+impl VrefintCal {
+    /// Read calibration value, acquired at temperature of 25 °C and V_DDA = 3 V.
```

Comment:
Maybe mention `VREFINT_CAL` somewhere which is the name used in the datasheet. Makes it easier to search for it in the documentation :slightly_smiling_face:

Reply:
Good point. I actually prefer to follow the naming convention from the official documentation where practical, even if that conflicts with Rust's naming convention. So I wouldn't mind if this struct were called `VREFINT_CAL`, if that's the "official" name.

Reply:
I'd prefer to stick to Rust naming for the struct, but I'll add a reference to `VREFINT_CAL` to the docs!